### PR TITLE
Starcraft 2 Wings of Liberty AP Implementation

### PR DESCRIPTION
**Pull request by TheCondor07**
Randomizes randomize when you obtain unit unlocks, armory upgrades, and laboratory researches. Features a modified text client to connect to AP and launch Starcraft 2.

### CODE REVIEW

**Comment by TheCondor07** [comment]
Seems like it wasn't set up for the new tutorial stuff, working on it now.

**Comment by TheCondor07** [comment]
Alright, tutorial has been properly implemented.

File: StarcraftClient.py [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+    sent_announce_pos = 0
+    announcements = []
+    announcement_pos = 0
+
+    def __init__(self, server_address, password):
+        super(Context, self).__init__(server_address, password)
+
+    async def connection_closed(self):
+        await super(Context, self).connection_closed()
+
```
Comment (Berserker66) [inline StarcraftClient.py:87]:
```suggestion

```

Since all these do is refer to the parent class, it cane be removed

File: StarcraftClient.py [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+                    game_state = int(38281 - unit.health)
+                    can_read_game = True
+
+            if iteration == 80 and not game_state & 1:
+                await self.chat_send("SendMessage Warning: Archipelago unable to connect or has lost connection to " +
+                                     "Starcraft 2 (This is likely a map issue)")
+
+            if self.last_received_update < len(self.ctx.items_received):
+                #for i in range(self.last_received_update, len(self.ctx.items_received))
+                #    await self.chat_send("SendMessage Received {}")
```
Comment (Berserker66) [inline StarcraftClient.py:307]:
```suggestion

```

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**